agent/tcs/client/client.go

```diff
@@ -67,7 +66,7 @@ func New(url string, region string, credentialProvider *credentials.Credentials,
 // call as unhandled requests will be discarded.
 func (cs *clientServer) Serve() error {
 	seelog.Debug("TCS client starting websocket poll loop")
-	if cs.Conn == nil {
+	if !cs.IsConnected() {
 		return fmt.Errorf("nil connection")
```

Comment:
The error message should be changed to indicate that it's not connected instead of `nil connection`